lib/generators/shopify_app/install/templates/shopify_app.rb.tt

```diff
@@ -1,16 +1,18 @@
-ShopifyApp.configure do |config|
-  config.application_name = "<%= @application_name %>"
-  config.api_key = ENV.fetch('SHOPIFY_API_KEY', '').presence || raise('Missing SHOPIFY_API_KEY')
-  config.secret = ENV.fetch('SHOPIFY_API_SECRET', '').presence || raise('Missing SHOPIFY_API_SECRET')
-  config.old_secret = "<%= @old_secret %>"
-  config.scope = "<%= @scope %>" # Consult this page for more scope options:
-                                 # https://help.shopify.com/en/api/getting-started/authentication/oauth/scopes
-  config.embedded_app = <%= embedded_app? %>
-  config.after_authenticate_job = false
-  config.api_version = "<%= @api_version %>"
-  config.shop_session_repository = 'Shop'
-  config.allow_jwt_authentication = <%= !with_cookie_authentication? %>
-  config.allow_cookie_authentication = <%= with_cookie_authentication? %>
+unless defined? Rails::Generators
```

Comment:
@andyw8  with this change, the reference of `ShopifyApp.configuration` in generator code(not template code) should also be removed, there is one at [home_controller_generator.rb](https://github.com/Shopify/shopify_app/blob/master/lib/generators/shopify_app/home_controller/home_controller_generator.rb#L35)
```ruby
      def embedded_app?
        ShopifyApp.configuration.embedded_app?
      end
```
the method `embedded_app?` is no longer needed any more. 
Otherwise the generated code of `HomeController` will be an authenticated home_controller instead of the unauthenticated home_controller, which is conflict with the [README.md](https://github.com/Shopify/shopify_app/blob/master/README.md#home-controller-generator)

>  By default, this generator creates an unauthenticated home_controller and a sample protected products_controller.

Reply:
That's a good point @xiaoboa, I'm actually investigating that myself. We still need need the method if we're creating non-embedded apps but I'll make sure that home_controller is not authenticated.

Thanks!